File: ProjectPO_Java/src/PO_java/LifeSymulator/Animals/Fox.java
package PO_java.LifeSymulator.Animals;

import PO_java.LifeSymulator.Animal;
import PO_java.LifeSymulator.Organism;
import PO_java.LifeSymulator.Tools.Cords;
import PO_java.LifeSymulator.World;

public class Fox extends Animal {
    public Fox(World world, Cords cords)
    {
        strength = 3;
        initiative = 7;
        age = 0;
        sign = 'F';
        name = "Fox";
        toKill = false;
        isDead = false;
        this.cords = cords;
        this.world = world;
    }

    @Override
    public void action(){
        int moveRange = 1;
        Cords currentCords = this.cords;
        Cords newCords;
        newCords = new Cords(-1, -1);
        while (!checkMove(newCords))
            newCords.setCords(randomNewCords(currentCords, moveRange));
        if (checkMove(newCords))
        {
            if (world.isSomethingThere(newCords))
            {
                Organism occupyingOrganism = world.whatIsThere(newCords);
                if(occupyingOrganism.getStrength() <= this.strength)
                    collision(occupyingOrganism);
            }
            else
                move(this, newCords, currentCords);
        }
    }
}

File: ProjectCppPO/Fox.cpp
#include "Fox.h"

Fox::Fox(World* world, Cords cords)
{
	features.strength = 3;
	features.initiative = 7;
	features.age = 0;
	features.look = 'F';
	features.name = "Fox";
	features.toKill = false;
	location = cords;
	this->world = world;
}

//Metoda nadpisuj¹ca metodê z klasy rodzica ¿eby lis nigdy nie wszed³ na pole z organizmem silniejszym od siebie
void Fox::action()
{
	int moveRange = 1;
	Cords currentCords = this->getLocation();
	Cords newCords(-1, -1);
	while (!checkMove(newCords))
		newCords.setCords(randomNewCords(currentCords, moveRange));
	if (checkMove(newCords))
	{
		if (world->isSomethingThere(newCords))
		{
			Organism* occupyingOrganism = world->whatIsThere(newCords);
			if(occupyingOrganism->getFeatures().strength <= this->features.strength)
				collision(occupyingOrganism);
		}
		else
			move(this, newCords, currentCords);
	}
}